app/uk/gov/hmrc/gform/gformbackend/GformConnector.scala

```diff
@@ -85,6 +85,38 @@ class GformConnector(ws: WSHttp, baseUrl: String) {
   def maybeForm(formIdData: FormIdData, formTemplate: FormTemplate)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
+  ): Future[Option[Form]] =
+    findForm(formIdData, formTemplate).recoverWith {
+      case UpstreamErrorResponse.WithStatusCode(statusCode, _) if statusCode == StatusCodes.NotFound.intValue =>
+        def getFormByLegacyIds(legacyIds: List[FormTemplateId]): Future[Option[Form]] =
+          legacyIds match {
+            case Nil => Future.successful(None)
+            case legacyFormId :: tail =>
+              getFormTemplate(legacyFormId)
```

Comment:
I still find it weird we are doing this. I think we shouldn't assume old form template is still present in db.

Only reason why you need full `FormTemplate` is to pass it `findForm` method, but I think we should not use `findForm` method for this legacy ids and use directly `getForm` instead.